### PR TITLE
removing dependency on httpd for cinde,glance,neutron,nova

### PR DESCRIPTION
**Pull request by saneax**
On Vagrant we were seeng oc1 (keystone) failure on dependency for (say) cinder,
which was again failing on dependency for httpd, and httpd was failing on dependency for keystone

### CODE REVIEW

**Comment by saneax** [comment]
please test it

**Comment by saneax** [comment]
please test this

**Comment by bodepd** [comment]
@saneax 

two problems with this patch that I can see: 

== missing dependency is not specified:

all keystone objects need to come only after Service[http] is started

== commit messages should have enough context for anyone reviewing code to understand:

* what was the problem (ie: what problem was occurring)
* what does the fix do

if I had not been sitting with you debugging, I could not really understand the intention of this patch without really digging into the code.

@hkumarmk for context when you review:

basically, there is a missing dependency between:

```
Service['http'] -> Keystone_obj<||>
```

in fact, there are deps to ensure that the keystone stuff has to come after keystone is installed b/c keystone_obj -> service-api -> http (and the keystone calls cannot work without http running)

for the case of acceptance, it only happened to work b/c the --tags package run from userdata happened to install apache.

**Comment by bodepd** [comment]
from a functional perspective, the patch does appear to result in a working cluster

**Comment by saneax** [comment]
please test this

**Comment by saneax** [comment]
The test has failed due to timeout, not running a new one.

**Comment by saneax** [comment]
please test this

**Comment by hkumarmk** [comment]
The reason why I put the order is  that, restarting apache with vhost configuration before (re)configuring api services will be failed, as  vhost will have same port configuration as api services (we are actually moving the default api ports to apache vhosts), will need to analyze this more to comment more on this.

**Comment by anshprat** [comment]
closing this as not required.
